jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/SpansToDependencyLinks.java

```diff
@@ -57,28 +59,19 @@
                 continue;
             }
 
-            // if current span is shared server span we don't want to create a link
-            // because the link should be for client span
-            if (existsSpanWhichStartedBefore(span, spanMap)) {
-                continue;
-            }
-
-            Set<OnlySpanIdsKey> parents = spanMap.get(span.getParentId());
+            Set<Span> parents = spanMap.get(span.getParentId());
             if (parents != null) {
                 if (parents.size() > 1) {
                     // multiple possible parents so parent is zipkin shared span
                     // therefore we choose the one with the greatest start time (e.g. server span)
                     // we cannot choose client span because it's not a true parent!
-                    Span sharedParent = null;
-                    for (OnlySpanIdsKey sharedSpan: parents) {
-                        if (sharedParent == null || sharedSpan.span.getStartTime() > sharedParent.getStartTime()) {
-                            sharedParent = sharedSpan.span;
-                        }
-                    }
-                    result.add(new Dependency(sharedParent.getProcess().getServiceName(), span.getProcess().getServiceName()));
+                    firstNotClientSpan(parents)
```

Comment:
rather than !clientSpan, wouldn't it be better to just get the serverSpan? in case there are spans that don't have either client or server annotations?

Reply:
+1